libraries/AP_GPS/GPS_Backend.cpp

```diff
@@ -287,5 +287,19 @@ void AP_GPS_Backend::check_new_itow(uint32_t itow, uint32_t msg_length)
         // use msg arrival time, and correct for jitter
         uint64_t local_us = jitter_correction.correct_offboard_timestamp_usec(_pseudo_itow, uart_us);
         state.uart_timestamp_ms = local_us / 1000U;
+
+        // look for lagged data from the GPS. This is meant to detect
+        // the case that the GPS is trying to push more data into the
+        // UART than can fit (eg. with GPS_RAW_DATA at 115200).
+        float expected_lag;
+        if (gps.get_lag(state.instance, expected_lag)) {
+            float lag_s = (now - state.uart_timestamp_ms) * 0.001;
```

Comment:
isn't it better in this case to use AP_HAL::millis() for better accuracy instead of now ? we some calculus with division, so better accuracy maybe obtain from getting fresh time